### PR TITLE
SUP-15062 - Implement new part type route, add tests, add changelog, add migration

### PR DESCRIPTION
**Pull request by guggi**
## Issues: 
- https://jira.gentics.com/browse/SUP-15062
- https://jira.gentics.com/browse/SUP-15298

## ToDo:

- javadoc :heavy_check_mark:

### CODE REVIEW

File: cms-core/src/main/resources/changelog/content.CHANGELOG [hunk 1]
```diff
@@ -786,3 +786,4 @@
 2023-03-30  CH-1694 NOP SQL DROP TABLE IF EXISTS page_processes;
 2023-03-30  CH-1695 NOP SQL DROP TABLE IF EXISTS page_task;
 2023-03-30  CH-1696 NOP SQL DROP TABLE IF EXISTS template_processes;
+2023-07-10  CH-1698 OG SQL ALTER TABLE type ADD deprecated tinyint(1);
```
Comment (guggi) [inline cms-core/src/main/resources/changelog/content.CHANGELOG:789]: is this file the proper location for adding the migration? 
is tinyint display width 1, fine?

File: cms-oss-doc/src/main/source/gcn_part_types.textile [hunk 2]
```diff
@@ -36,8 +36,8 @@ h3. Overview
 | 31        | Checkbox                | Part type for input of boolean value. |
 | 32        | Datasource              | With this part type, the user can define individual datasources in the tag. |
 | 33        | Velocity                | The velocity part type can be used to evaluate all previously generated text using the velocity markup parser. |
-| 34        | -Breadcrumb-            | This part type can be used to generate breadcrumbs. (deprecated) |
-| 35        | -Navigation-            | The navigation part type can be used to create a custom navigation using a velocity template. (deprecated) |
+| 34        | Breadcrumb            | This part type can be used to generate breadcrumbs. |
+| 35        | Navigation            | The navigation part type can be used to create a custom navigation using a velocity template. |
```
Comment (guggi) [inline cms-oss-doc/src/main/source/gcn_part_types.textile:40]: Should the de-deprecation be mentioned somewhere else?

File: cms-core/src/main/resources/changelog/content.CHANGELOG (hunk 1 above)
Comment (npomaroli) [reply cms-core/src/main/resources/changelog/content.CHANGELOG:789]: The file is correct. As for the datatype, you could use BOOLEAN, and you should set to to NOT NULL and add a default value.
So the complete SQL should be
ALTER TABLE type ADD COLUMN deprecated BOOLEAN NOT NULL DEFAULT 0;

File: cms-restapi/src/main/java/com/gentics/contentnode/rest/model/PartType.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    this.id = rs.getInt("id");
+    this.name = rs.getString("name");
+    this.description = rs.getString("description");
+    this.auto = rs.getBoolean("auto");
+    this.javaClass = rs.getString("javaclass");
+    this.deprecated = rs.getBoolean("deprecated");
+    return this;
+  }
+
+  public String getName() {
```
Comment (yrucrem) [inline cms-restapi/src/main/java/com/gentics/contentnode/rest/model/PartType.java:28]: Add JavaDoc for getters & setters.

File: cms-restapi/src/main/java/com/gentics/contentnode/rest/resource/PartTypeResource.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+public abstract class PartTypeResource {
+
+	/**
+	 * List Part types
+	 * @param filter filter parameters
+	 * @return list of part type
+	 * @throws Exception in case of errors
+	 */
+	public abstract List<PartType> list(
```
Comment (yrucrem) [inline cms-restapi/src/main/java/com/gentics/contentnode/rest/resource/PartTypeResource.java:16]: Add @GET & @Path annotations.

File: cms-oss-changelog/src/changelog/entries/2023/07/7470.SUP-15062.enhancement [hunk 5]
```diff
@@ -0,0 +1 @@
+Adding new endpoint *@GET /part* to retrieve all kind of part types (including custom part types).
```
Comment (yrucrem) [inline cms-oss-changelog/src/changelog/entries/2023/07/7470.SUP-15062.enhancement:1]: The endpoint should be @GET /parttype@ I think. The @ are for the code formatting, and according to the resource implementation the path is /parttype

**Review by yrucrem — CHANGES_REQUESTED** [review]

File: cms-core/src/main/resources/changelog/content.CHANGELOG (hunk 1 above)
Comment (guggi) [reply cms-core/src/main/resources/changelog/content.CHANGELOG]: d45de5b

File: cms-restapi/src/main/java/com/gentics/contentnode/rest/model/PartType.java (hunk 3 above)
Comment (guggi) [reply cms-restapi/src/main/java/com/gentics/contentnode/rest/model/PartType.java:35]: d45de5b

File: cms-oss-changelog/src/changelog/entries/2023/07/7470.SUP-15062.enhancement (hunk 5 above)
Comment (guggi) [reply cms-oss-changelog/src/changelog/entries/2023/07/7470.SUP-15062.enhancement]: nice catch! :pray:  i already forgot the renaming :see_no_evil:

**Pull request merged**